coroutine.go

```diff
@@ -53,12 +58,15 @@ type Context[R, S any] struct {
 	recv R
 	send S
 
+	// Value returned from the coroutine.
+	result R
```

Comment:
Could we make the result parameter a different type than R? Or do you think a coroutine should only return the same type than it yields?

Reply:
I think of a yield as a type of return. We already have cases upstream where there's more than one type of yield that each needs to pass different types of information back to the process scheduling coroutines. To work around the fact that we can't have a type parameter per yield type (because the number of unique yield types is unknown), the type `R` is likely a compound type that can carry different types of information.

This PR introduces another type of return/yield. We can reuse `R` to keep things simple, or we could introduce a new type parameter to tighten the contract a bit. I opted to keep it simple and reuse `R` for now, since we can always split out a new type parameter later. Reusing `R` means that we don't need to introduce a new type parameter on `coroutine.Yield` (and others), which would be a breaking change, would require changes to the compiler, and which would couple all `Yield` points to the coroutine's return value.

I don't feel strongly either way so if you feel like the extra type safety is worth it I can make the change.

Reply:
Let's start with this if you think it's good enough, until we feel like it's starting to be too restrictive 👍